terms/neuralnetwork_bp.tex

```diff
@@ -0,0 +1,75 @@
+		\caption{The Tanh function.}
+		\label{fig:tanhfig}
+	\end{marginfigure}
+	The Hyperbolic Tangent function\index{Hyperbolic Tangent function}\index{Tanh} (Tanh) is very similar to the Sigmoid function, with a more stable gradient and an asymptotic range between -1 and 1, thus including negative values \cite{Haykin:1994:NNC:541500,sharmav_2017}. The function shown in figure~\ref{fig:tanhfig} is the following \cite{sharmav_2017}: $$tanh(x) = \frac{2}{1+e^{-2x}}-1 $$
+	
+\end{itemize}
+The traversal goes from input nodes to the output nodes, and the output will be a mapped function of the input, firing activation functions\index{activation function} at each step \cite{GURESEN2011426, sharmav_2017}.\\ Before performing any predictions, the ANN needs to be trained with labelled data in order to adjust the synapses to their optimal values to form the mapping. \\ The goal of the learning process is to minimize the output of a cost function\index{cost function}, taking into consideration the error value \cite{Haykin:1994:NNC:541500}. Each time the input is fed forward through the network, the error of the predicted output is compared to the actual output and thus produce an error value. The cost function $C(n)$, in terms of error $e_k(n)$ is defined as $C(n) = \frac{1}{2}e^2_k(n)$ \cite{Haykin:1994:NNC:541500}. This error needs to be minimized, which is done through backpropagation\index{backpropagation}, which is the process where the biases are updated by distributing the error from the layer closest to the output inwards, applying the derivative of the activation function in the process \cite{doi:10.1080/02626669809492102}.\\ The error delta in terms of weight $w_{kj}$ is calculated with the following formula: $\Delta w_{kj}(n) = \alpha e_k(n)x_j(n)$ where $\alpha$ is the learning rate\index{learning rate}, $e_k(n)$ is the error and $x_j(n)$ is the input \cite{Haykin:1994:NNC:541500}. The synapses are updated by adding the calculated delta to the weight, formally, $w_{kj}(n+1) = w_{kj}(n) + \Delta w_{kj}(n)$ where $w_{kj}(n)$ is the previous weight and $\Delta w_{kj}(n)$ is the calculated weight difference \cite{Haykin:1994:NNC:541500}.\\ Techniques such as gradient descent are used to converge the weights to a minimum error value at a specified rate of change, called the learning rate \cite{doi:10.1080/02626669809492102}.\\ As the ANN performs epochs, the error of the synapses will converge towards the optimal value, which once reached, the model will be trained. Once the model is trained, one can perform predictions by passing an input through the network and observing the output.
+\begin{figure}
+	\includegraphics{neural_network/ANN}
+	%  \checkparity This is an \pageparity\ page.%
+	\caption[ANN][6pt]{This diagram is a visual representation of an ANN.}
+	\label{fig:annfig}
+	%\zsavepos{pos:annfig}
+\end{figure}
+
+\subsection{Convolutional Neural Network}\label{sec:headings}\index{headings}
```

Comment:
\index{headings} doesn't make any sense.  WHo is going to look for "Headings" in the index???  see guidelines 11 and 21